config-controller/api/v1alpha1/policy_types.go

```diff
@@ -38,31 +38,47 @@ type EnforcementAction string
-	Exclusions      []Exclusion      `json:"exclusions,omitempty"`
-	Scope           []Scope          `json:"scope,omitempty"`
+	// LifecycleStages describes which policy lifecylce stages this policy applies to.  Choices are DEPLOY, BUILD, and RUNTIME.
+	LifecycleStages []LifecycleStage `json:"lifecycleStages"`
+	// EventSource describes which events should trigger execution of this policy
+	EventSource EventSource `json:"eventSource,omitempty"`
+	// Exclusions define deployments or images that should be excluded from this policy.
+	Exclusions []Exclusion `json:"exclusions,omitempty"`
+	// Scope defines clusters, namespaces, and deployments that should be included in this policy.  No scopes defined includes everything.
+	Scope []Scope `json:"scope,omitempty"`
 	// +kubebuilder:validation:Required
 	// +kubebuilder:validation:Enum=UNSET_SEVERITY;LOW_SEVERITY;MEDIUM_SEVERITY;HIGH_SEVERITY;CRITICAL_SEVERITY
-	Severity           string              `json:"severity,omitempty"`
+	// Severity defines how severe a violation from this policy is.  Possible values are UNSET_SEVERITY, LOW_SEVERITY, MEDIUM_SEVERITY, HIGH_SEVERITY, and CRITICAL_SEVERITY.
+	Severity string `json:"severity"`
+	// Enforcement lists the enforcement actions to take when a violation from this policy is identified.  Possible value are UNSET_ENFORCEMENT, SCALE_TO_ZERO_ENFORCEMENT, UNSATISFIABLE_NODE_CONSTRAINT_ENFORCEMENT, KILL_POD_ENFORCEMENT, FAIL_BUILD_ENFORCEMENT, FAIL_KUBE_REQUEST_ENFORCEMENT, FAIL_DEPLOYMENT_CREATE_ENFORCEMENT, and. FAIL_DEPLOYMENT_UPDATE_ENFORCEMENT.
 	EnforcementActions []EnforcementAction `json:"enforcementActions,omitempty"`
-	Notifiers          []string            `json:"notifiers,omitempty"`
+	// Notifiers is a list of notifier IDs that should be triggered when a violation from this policy is identified.  IDs should be in the form of a UUID and are found in the Central API.
+	Notifiers []string `json:"notifiers,omitempty"`
 	// +kubebuilder:validation:MinItems=1
-	PolicySections     []PolicySection      `json:"policySections,omitempty"`
+	// PolicySections define the violation criteria for this policy.
+	PolicySections     []PolicySection      `json:"policySections"`
 	MitreAttackVectors []MitreAttackVectors `json:"mitreAttackVectors,omitempty"`
-	CriteriaLocked     bool                 `json:"criteriaLocked,omitempty"`
-	MitreVectorsLocked bool                 `json:"mitreVectorsLocked,omitempty"`
-	IsDefault          bool                 `json:"isDefault,omitempty"`
+	// Read-only field. If true, the policy's criteria fields are rendered read-only.
+	CriteriaLocked bool `json:"criteriaLocked,omitempty"`
+	// Read-only field. If true, the policy's MITRE ATT&CK fields are rendered read-only.
+	MitreVectorsLocked bool `json:"mitreVectorsLocked,omitempty"`
+	// Read-only field. Indicates the policy is a default policy if true and a custom policy if false.
+	IsDefault bool `json:"isDefault,omitempty"`
```

Comment:
@kylape @clickboo @c-du 

TODO:  come up with a plan for the future of default policies. It's weird to allow a customer to designate a bespoke, custom policy a "default" when we use the term to mean "comes out-of-the-box when you install ACS".

Reply:
https://issues.redhat.com/browse/ROX-26757